mobile/native/src/max_quantity.rs

```diff
@@ -33,16 +35,75 @@ pub fn max_quantity(
         }
     };
 
-    let max_coordinator_margin =
+    let max_coordinator_balance =
         Amount::from_sat(channel_trade_constraints.max_counterparty_balance_sats);
-    let max_trader_margin = Amount::from_sat(channel_trade_constraints.max_local_balance_sats);
+    let max_trader_balance = Amount::from_sat(channel_trade_constraints.max_local_balance_sats);
     let order_matching_fee_rate = channel_trade_constraints.order_matching_fee_rate;
-    let order_matching_fee_rate =
-        Decimal::try_from(order_matching_fee_rate).expect("to fit into decimal");
+    let order_matching_fee_rate = Decimal::try_from(order_matching_fee_rate).expect("to fit");
 
-    let accumulated_order_matching_fees = match position::handler::get_positions()?.first() {
-        Some(position) => position.order_matching_fees,
-        _ => Amount::ZERO,
+    let positions = position::handler::get_positions()?;
+    let position = positions.first();
+    let accumulated_order_matching_fees = position
+        .map(|p| p.order_matching_fees)
+        .unwrap_or(Amount::ZERO);
+
+    let (max_coordinator_margin, max_trader_margin) = match position {
+        Some(position) if trader_direction != position.direction => {
```

Comment:
🙃 `trader_direction` is slightly ambiguous. I think it should be `trade_direction`.

Reply:
But it's the direction that the trader has chosen. I think `trade_direction` wouldn't be clear enough.

Reply:
`trader_direction` is ambiguous because the trader has both a trade/order direction and a position direction. If you want to be very specific I think it should be `trader_trade_direction` or `trader_order_direction`.

Reply:
I changed it to `trader_trade_direction`